api/dto/subscription.go

```diff
@@ -83,11 +82,8 @@ func (subscription *Subscription) checkContacts(request *http.Request) error {
 			return ErrProvidedContactsForbidden{contactIds: anotherUserContactIds}
 		}
 		anotherUserNames := make([]string, 0)
-		anotherContactIds := make([]string, 0)
-		for i, contact := range contacts {
-			if contact == nil {
-				anotherContactIds = append(anotherContactIds, anotherUserContactIds[i])
-			} else {
+		for _, contact := range contacts {
+			if contact != nil {
```

Comment:
Actually, that is a bug. `anotherContactIds` should be used in line 92 (or 96 in the original file) instead of `anotherUserContactIds`.